Merged domain into usecases module. Added basic Unit Test.

matuki opened the pull request:
- Merged domain and usecases modules into usecases
- Removed dependency of usecases from data using dependency inversion
- Added a basic unit test for GetWallpapers usecase

Commit:
- Merged domain and usecases modules into usecases
- Removed dependency of usecases from data using dependency inversion
- Added a basic testcase for GetWallpapers usecase

igorescodro commented on app/src/test/java/com/pmatuki/wowpapers/fake/WallpaperListSourceFake.kt:
```diff
@@ -0,0 +1,37 @@
+import com.pmatuki.wowpapers.model.Thumb
+import com.pmatuki.wowpapers.model.Wallpaper
+
+class WallpaperListSourceFake(val wallpaperCount: Int) : WallpaperListSource {
```
Yes, that's the idea! 😊

In your fake class you can have some "setup functions" in order to test multiple behavior scenarios.
For example, you can have a "throwException" function that when you call "getWallpaperList()" will fail, you can have a "returnEmptyList" or else. 

Then your test will look something like:

```
@Test(expected = IllegalArgumentException::class)
fun `test if error is returned`()  = runBlockingTest {
    getWallpapers.throwError()
    val wallpaperList = getWallpapers()
}
```

You can do what you want with a Fake class and reuse in other tests without needing to mocking everything over and over again.

igorescodro commented on app/src/test/java/com/pmatuki/wowpapers/usecases/GetWallpapersTest.kt:
```diff
@@ -0,0 +1,28 @@
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `test if get wallpapers use case returns expected list`()  = runBlockingTest {
```
Like a comment above, maybe in this test is a good idea to validate different states like possible errors, empty lists and any other corner case this API may return.

Commit:
- Added test cases to "list wallpapers" test.
- Added "state" to WallpaperListSourceFake so different scenarios can be tested.
